src/ui/public/stringify/__tests__/_duration.js

```diff
@@ -0,0 +1,85 @@
+import expect from 'expect.js';
+import ngMock from 'ngMock';
+import RegistryFieldFormatsProvider from 'ui/registry/field_formats';
+
+describe('Duration Format', function () {
+  var fieldFormats;
+  var DurationFormat;
+
+  beforeEach(ngMock.module('kibana'));
+  beforeEach(ngMock.inject(function (Private) {
+    fieldFormats = Private(RegistryFieldFormatsProvider);
+    DurationFormat = fieldFormats.getType('duration');
+  }));
+
+  const inputs = [
+    -60,
+    60,
+    125
+  ];
+  const fixtures = [{
+    params: {
+      inputFormat: 'seconds',
+      outputFormat: 'humanize'
+    },
+    outputs: [
+      'minus a minute',
+      'a minute',
+      '2 minutes'
+    ]
+  }, {
+    params: {
+      inputFormat: 'seconds',
+      outputFormat: 'asSeconds',
+      outputPrecision: 0
+    },
+    outputs: [
+      '-60',
+      '60',
+      '125'
+    ]
+  }, {
+    params: {
+      inputFormat: 'seconds',
+      outputFormat: 'asSeconds',
+      outputPrecision: 2
+    },
+    outputs: [
+      '-60.00',
+      '60.00',
+      '125.00'
+    ]
+  }, {
+    params: {
+      inputFormat: 'minutes',
+      outputFormat: 'asHours',
+      outputPrecision: 0
+    },
+    outputs: [
+      '-1',
+      '1',
+      '2'
+    ]
+  }, {
+    params: {
+      inputFormat: 'minutes',
+      outputFormat: 'humanize'
+    },
+    outputs: [
+      'minus an hour',
+      'an hour',
+      '2 hours'
+    ]
+  }];
+
+  fixtures.forEach((fixture, i) => {
```

Comment:
Avoiding repetition is good, but clarity and expressiveness should be stronger concerns within tests.  With some changes (namely not automating the test execution), you can still avoid duplicating the test case itself for common inputs/outputs and have expressive and concise tests.

One possible example of the same test cases: https://gist.github.com/epixa/aed0d611e41ac366073d

That all said, I'm pretty squarely in the don't-automate-test-creation camp, but there are others on the team that are more comfortable with it, so don't feel obligated to switch to the other extreme.